src/custom/state/claim/hooks/index.ts

```diff
@@ -279,6 +279,17 @@ export function useUserClaims(account: Account, optionalChainId?: SupportedChain
   return { claims: claimKey ? claimInfo[claimKey] : null, isLoading }
 }
 
+let FETCH_DEPLOYMENT_TIMESTAMP_PROMISE: Promise<number> | null = null
+function fetchDeploymentTimestamp(vCowContract: VCowType) {
+  return (
+    FETCH_DEPLOYMENT_TIMESTAMP_PROMISE ??
+    (FETCH_DEPLOYMENT_TIMESTAMP_PROMISE = vCowContract.deploymentTimestamp().then((ts) => {
```

Comment:
Can we use await?

Reply:
I would keep then here.

Reply:
Not against it, but any reason in particular? 
it seems more readable to use await